managementApp/src/main/java/com/skytask/store/ProductStore.java

```diff
@@ -0,0 +1,35 @@
+package com.skytask.store;
+
+import com.skytask.common.Product;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+@Component
+public class ProductStore {
+
+    private List<Product> products;
```

Comment:
Please use a different approach, this one has concurrency problems.

Reply:
But I have synchronized methods, I read that it makes it able to work multi-thread.

Reply:
`Thread 1. Petición GET a /
Thread 1. Lanza petición a Rabbit a getProductsList
Thread 1. Solicita lista de productos con productStore.getProducts() (entra al ciclo donde espera por el flag)
Thread 2. Petición GET a / 
Thread 2. Lanza petición a Rabbit a getProductsList
Thread 2. Solicita lista de productos con productStore.getProducts() (se bloquea por el synchronized)
Thread 3. Petición GET a / 
Thread 3. Lanza petición a Rabbit a getProductsList
Thread 3. Solicita lista de productos con productStore.getProducts() (se bloquea por el synchronized)

Thread 4. Regresa lista solicitada por Thread 1 (llamemos lista1)
Thread 4. Establece la lista1 y pone flag a true
Thread 4. Regresa lista solicitada por Thread 2 (llamemos lista2)
Thread 4. Establece la lista2 y pone flag a true
Thread 4. Regresa lista solicitada por Thread 3 (llamemos lista3)
Thread 4. Establece la lista3 y pone flag a true

Thread 1. Ve que flag es true y regresa la lista3, establece flag a false
Thread 3. Despierta y entra a ciclo para esperar a que flag sea true
Thread 2. Sigue bloqueado`